src/main/java/thePackmaster/ui/EnhanceBonfireOption.java

```diff
@@ -0,0 +1,94 @@
+import thePackmaster.packs.GemsPack;
+import thePackmaster.vfx.gemspack.SocketGemEffect;
+
+
+public class EnhanceBonfireOption extends AbstractCampfireOption {
+    public static final String[] DESCRIPTIONS;
+    private static final UIStrings UI_STRINGS;
+
+    static {
+        UI_STRINGS = CardCrawlGame.languagePack.getUIString(SpireAnniversary5Mod.makeID("EnhanceBonfireOptions"));
+        DESCRIPTIONS = UI_STRINGS.TEXT;
+
+    }
+
+    //private ArrayList<String> idleMessages;
+    public EnhanceBonfireOption(boolean active) {
+        this.label = DESCRIPTIONS[0];
+
+        for (AbstractCard c : AbstractDungeon.player.masterDeck.group) {
+            c.update();
+        }
+
+        this.usable = active;
+        if (active) {
+            this.description = DESCRIPTIONS[1];
+            this.img = ImageMaster.loadImage(SpireAnniversary5Mod.makeImagePath("ui/scrapcampfire.png"));
+        } else {
+            this.img = ImageMaster.loadImage(SpireAnniversary5Mod.makeImagePath("ui/scrapcampfiredisabled.png"));
+            if (AbstractDungeon.player.gold < 10){
+                this.description = DESCRIPTIONS[2];
+            } else {
+                this.description = DESCRIPTIONS[2];
+            }
+        }
+    }
+
+    @Override
+    public void useOption() {
+        if (this.usable) {
+            GemsPack.currSocketGemEffect = new SocketGemEffect();
+            AbstractDungeon.effectList.add(GemsPack.currSocketGemEffect);
+
+        }
+    }
+
+    public void reCheck() {
+        if (SocketGemEffect.getModifiableCards().size() == 0) {
+            this.usable = false;
+        }
+        if (SocketGemEffect.getGems().size() == 0) {
+            this.usable = false;
+        }
+        if (this.usable) {
+            this.description = DESCRIPTIONS[1];
+            this.img = ImageMaster.loadImage(SpireAnniversary5Mod.makeImagePath("ui/scrapcampfire.png"));
+        } else {
+            this.description = DESCRIPTIONS[2];
+            this.img = ImageMaster.loadImage(SpireAnniversary5Mod.makeImagePath("ui/scrapcampfiredisabled.png"));
+        }
+
+    }
+
+    @Override
+    public void update() {
+        float hackScale = (float) ReflectionHacks.getPrivate(this, AbstractCampfireOption.class, "scale");
+
+        if (this.hb.hovered) {
+
+            if (!this.hb.clickStarted) {
+                ReflectionHacks.setPrivate(this, AbstractCampfireOption.class, "scale", MathHelper.scaleLerpSnap(hackScale, Settings.scale));
+                ReflectionHacks.setPrivate(this, AbstractCampfireOption.class, "scale", MathHelper.scaleLerpSnap(hackScale, Settings.scale));
+
+            } else {
+                ReflectionHacks.setPrivate(this, AbstractCampfireOption.class, "scale", MathHelper.scaleLerpSnap(hackScale, 0.9F * Settings.scale));
+
+            }
+        } else {
+            ReflectionHacks.setPrivate(this, AbstractCampfireOption.class, "scale", MathHelper.scaleLerpSnap(hackScale, 0.9F * Settings.scale));
```

Comment:
What does all this do?

Reply:
This allows the button to be mouse over even when you don't meet the requirements, so you can see the tooltip.  The scale change mimics what normally happens when the button is active.

Reply:
Normally greyed out options lose their hitbox, and we wanted to have this one be able to tell the player why its greyed out.